src/Bicep.LangServer/Handlers/BicepDeploymentScopeRequestHandler.cs

```diff
@@ -0,0 +1,155 @@
+            this.moduleDispatcher = moduleDispatcher;
+            this.namespaceProvider = namespaceProvider;
+        }
+
+        public Task<BicepDeploymentScopeResponse> Handle(BicepDeploymentScopeParams request, CancellationToken cancellationToken)
+        {
+            var documentUri = request.TextDocument.Uri;
+
+            Compilation? compilation;
+
+            try
+            {
+                compilation = GetCompilation(documentUri);
+                var deploymentScope = GetDeploymentScope(compilation.GetEntrypointSemanticModel().TargetScope);
+
+                return Task.FromResult(new BicepDeploymentScopeResponse(deploymentScope, GetCompiledFile(compilation, documentUri), null));
+            }
+            catch (Exception exception)
+            {
+                return Task.FromResult(new BicepDeploymentScopeResponse(ResourceScope.None.ToString(), null, exception.Message));
+            }
+        }
+
+        private static string GetDeploymentScope(ResourceScope resourceScope)
+        {
+            switch (resourceScope)
+            {
+                case ResourceScope.ResourceGroup:
+                    return LanguageConstants.TargetScopeTypeResourceGroup;
+                case ResourceScope.Subscription:
+                    return LanguageConstants.TargetScopeTypeSubscription;
+                case ResourceScope.ManagementGroup:
+                    return LanguageConstants.TargetScopeTypeManagementGroup;
+                case ResourceScope.Tenant:
+                    return LanguageConstants.TargetScopeTypeTenant;
+                default:
+                    return resourceScope.ToString();
+            }
+        }
+
+        private string GetCompiledFile(Compilation compilation, DocumentUri documentUri)
+        {
+            var fileUri = documentUri.ToUri();
+
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == fileUri);
+
+            if (diagnosticsByFile.Value.Any(x => x.Level == DiagnosticLevel.Error))
+            {
+                throw new Exception(DiagnosticsHelper.GetDiagnosticsMessage(diagnosticsByFile));
+            }
+
+            var stringBuilder = new StringBuilder();
+            var stringWriter = new StringWriter(stringBuilder);
+
+            var emitter = new TemplateEmitter(compilation.GetEntrypointSemanticModel(), emitterSettings);
+            emitter.Emit(stringWriter);
+
+            return stringBuilder.ToString();
+        }
+
+        private Compilation GetCompilation(DocumentUri documentUri)
+        {
+            var fileUri = documentUri.ToUri();
+            RootConfiguration? configuration;
+
+            try
+            {
+                configuration = this.configurationManager.GetConfiguration(fileUri);
+            }
+            catch (ConfigurationException exception)
+            {
+                throw exception;
```

Comment:
I would suggest rethrowing the exception implicitly to preserve stack details, even if we may not need them:

```C#
catch (ConfigurationException)
{
    throw;
}
```

Reply:
Updated here:
https://github.com/Azure/bicep/commit/5175955691bc0a1ca06efa73c481c10a2acd315c

Thanks!